### PR TITLE
Parallelizing the jballerina scheduler

### PR DESCRIPTION
**Pull request by manuranga**

### CODE REVIEW

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal [hunk 1]
```diff
@@ -819,7 +819,8 @@ function generateMainMethod(bir:Function userMainFunc, jvm:ClassWriter cw, bir:P
 
     mv.visitTypeInsn(NEW, SCHEDULER);
     mv.visitInsn(DUP);
-    mv.visitMethodInsn(INVOKESPECIAL, SCHEDULER, "<init>", "()V", false);
+    mv.visitInsn(ICONST_4);
```
Comment (vinok88) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal]: Don't we need to have some sort of logic or a way to configure number of threads? How about codegening this to at-least get it from a utility method for the moment.

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/ParallelismTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.ballerinalang.test.jvm;
```
Comment (vinok88) [inline tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/ParallelismTest.java]: license header missing.

Comment (manuranga) [reply tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/ParallelismTest.java]: thanks fixed

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal (hunk 1 above)
Comment (manuranga) [reply compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal]: Will refactor in next pr. Created #15307 to track this.

**Pull request merged**